modules/bloop-rifle/src/main/scala/scala/build/blooprifle/internal/Operations.scala

```diff
@@ -327,4 +333,43 @@ object Operations {
     )
   }
 
+  def about(
+    host: String,
+    port: Int,
+    workingDir: Path,
+    in: InputStream,
+    out: OutputStream,
+    err: OutputStream,
+    logger: BloopRifleLogger
+  ): Int = {
+
+    val stop0         = new AtomicBoolean
+    val nailgunClient = TcpClient(host, port)
+    val streams       = Streams(in, out, err)
+
+    timeout(5 seconds) {
+      nailgunClient.run(
+        "about",
+        Array.empty,
+        workingDir,
+        sys.env.toMap,
+        streams,
+        logger.nailgunLogger,
+        stop0,
+        interactiveSession = false
+      )
+    }
+
+  }
+
+  def timeout[T](duration: Duration)(body: => T) = {
+    import scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
We shouldn't use the default `ExecutionContext` for that (it's rarely a good idea to use this one, it's assumptions sometimes change from a Scala version to another, …)

Some `Operations` methods already accept a `ScheduledExecutorService`. Code along those lines ought to work:
```scala
val p = Promise[Int]()
scheduler.schedule { () =>
  val retCode = nailgunClient.run(…)
  p.tryComplete(Success(retCode))
}
scheduler.schedule(() => { p.tryComplete(Failure(new TimeoutException(…)))  }, 5, MINUTES)
```
(We could even keep the `scheduler.*` calls values, and try to cancel the other when one completes.)

Reply:
Done